components/background_hang_monitor/sampler_linux.rs

```diff
@@ -209,33 +167,17 @@ impl Sampler for LinuxSampler {
                 .wait_through_intr()
                 .expect("msg2 failed");
 
-            let context = CONTEXT.load(Ordering::SeqCst);
-            let mut cursor = mem::MaybeUninit::uninit();
-            let ret = unsafe { unw_init_local(cursor.as_mut_ptr(), context) };
-            result = if ret == UNW_ESUCCESS {
-                let mut native_stack = NativeStack::new();
-                #[allow(clippy::while_let_loop)] // False positive
-                loop {
-                    let ip = match get_register(cursor.as_mut_ptr(), RegNum::Ip) {
-                        Ok(ip) => ip,
-                        Err(_) => break,
-                    };
-                    let sp = match get_register(cursor.as_mut_ptr(), RegNum::Sp) {
-                        Ok(sp) => sp,
-                        Err(_) => break,
-                    };
-                    if native_stack
-                        .process_register(ip as *mut _, sp as *mut _)
-                        .is_err() ||
-                        !step(cursor.as_mut_ptr()).unwrap_or(false)
-                    {
-                        break;
-                    }
-                }
-                Ok(native_stack)
-            } else {
-                Err(())
+            let mut native_stack = NativeStack::new();
+            unsafe {
+                backtrace::trace_unsynchronized(|frame| {
+                    let ip = frame.ip();
+                    let sp = frame.sp();
+
+                    //This return value here determines whether we proceed to the next stack frame or not.
```

Comment:
Nit:
```suggestion
                    // This return value here determines whether we proceed to the next stack frame or not.
```

Reply:
If `rustfmt` has an option to catch that, I would love to turn it on! after a quick glance at mhttps://github.com/rust-lang/rustfmt/blob/master/Configurations.md, I don't see it though.